### DISCUSSION TITLE
FINN Used for Education : a 9 hours course.

### DISCUSSION THREAD

**Discussion by 0BAB1** [Xilinx/finn#1204] [Show and tell]
Hello everyone,

As my previous posts and requests implied, I was working on a FINN project :

## A 9 hours  hands-on course on how to use FINN to deploy end-to end solution

Thanks to this amazing tool, I was able to teach in 9 hours key skills to deploy an end-to-end solution on Zynq, key skills including, but not limited to :

- Using Brevitas to train our own model, if needed
- Using FINN and understanding the basics of hardware inference
- Use the resulting stitched IP for custom FPGA integration

This is done through :
- 3 hours of lectures based on various resources, including the FINN papers
- 6 hours of labs, heavily inspired from the example notebooks, trimmed to the minimum and adapted to the example FashionMNIST classifier used for the course

I wanted to make this post to thank the researchers and show them that the tool is used for education !

The goal of the course was to kickstart student on their projects in a minimal amount of time thanks to FINN's ease of use, and also de-mystify the process by sharing this tool to everyone interested.

I taught this course last week for SKKU, got only good feedback so far and I will keep you updated on that as students will soon provide feedback too.

You can find more details about the course here : [Course Github](https://github.com/0BAB1/python_to_fpga_course).

**Comment by auphelia** [comment]
Thank you for creating this Show & Tell, @0BAB1! It's very exciting to see that you put a course on FINN & Brevitas together :)
